src/her.cc

```diff
@@ -58,99 +74,99 @@ void her(
     blas_error_if( lda < n );
     blas_error_if( incx == 0 );
 
-    // check for overflow in native BLAS integer type, if smaller than int64_t
-    if (sizeof(int64_t) > sizeof(blas_int)) {
-        blas_error_if( n              > std::numeric_limits<blas_int>::max() );
-        blas_error_if( lda            > std::numeric_limits<blas_int>::max() );
-        blas_error_if( std::abs(incx) > std::numeric_limits<blas_int>::max() );
-    }
-
-    blas_int n_    = (blas_int) n;
-    blas_int lda_  = (blas_int) lda;
-    blas_int incx_ = (blas_int) incx;
+    // convert arguments
+    blas_int n_    = to_blas_int( n );
+    blas_int lda_  = to_blas_int( lda );
+    blas_int incx_ = to_blas_int( incx );
 
-    // if x2=x, then it isn't modified
-    std::complex<float> *x2 = const_cast< std::complex<float>* >( x );
+    // Deal with layout. RowMajor needs copy of x in x2;
+    // in other cases, x2 == x.
+    scalar_t* x2 = const_cast< scalar_t* >( x );
     if (layout == Layout::RowMajor) {
         // swap lower <=> upper
         uplo = (uplo == Uplo::Lower ? Uplo::Upper : Uplo::Lower);
 
         // conjugate x (in x2)
-        x2 = new std::complex<float>[n];
+        x2 = new scalar_t[ n ];
         int64_t ix = (incx > 0 ? 0 : (-n + 1)*incx);
         for (int64_t i = 0; i < n; ++i) {
-            x2[i] = conj( x[ix] );
+            x2[ i ] = conj( x[ ix ] );
             ix += incx;
         }
         incx_ = 1;
     }
-
     char uplo_ = uplo2char( uplo );
-    BLAS_cher( &uplo_, &n_,
-               &alpha,
-               (blas_complex_float*) x2, &incx_,
-               (blas_complex_float*) A, &lda_ );
+
+    // call low-level wrapper
+    internal::her( uplo_, n_,
```

Comment:
Just thinking. Should not it be like:
```
internal::her(
  uplo_, [....] );
```

Reply:
The style guide allows either
```
foobar( a, b, c,
        x, y, z );
```
or
```
foobar(
    a, b, c,
    x, y, z );
```
if that's what you're asking about. For prototypes, I prefer the later, but for calls I use either one.